crates/sui-data-ingestion/src/reader.rs

```diff
@@ -112,17 +118,15 @@ impl LocalReader {
             .watch(&self.path, RecursiveMode::NonRecursive)
             .expect("Inotify watcher failed");
 
-        checkpoint_number = self
```

Comment:
removing the initial read when the daemon starts. If too many files accumulate in the directory, it can clog the channel between the reader and the executor causing a deadlock. To maintain a similar behavior as before during startup, a timeout has been added to the inotify channel